runtime/compiler/x/codegen/J9AheadOfTimeCompile.cpp

```diff
@@ -253,6 +252,23 @@ uint8_t *J9::X86::AheadOfTimeCompile::initializeAOTRelocationHeader(TR::Iterated
             resolvedMethod = guard->getSymbolReference()->getSymbol()->getResolvedMethodSymbol()->getResolvedMethod();
             }
 
+         if (comp->getOption(TR_UseSymbolValidationManager))
+            {
+            TR_OpaqueMethodBlock *method = resolvedMethod->getPersistentIdentifier();
+            TR_OpaqueClassBlock *thisClass = guard->getThisClass();
+            uint16_t methodID = symValManager->getIDFromSymbol(static_cast<void *>(method));
+            uint16_t receiverClassID = symValManager->getIDFromSymbol(static_cast<void *>(thisClass));
+
```

Comment:
Just checking why 16 bits for the ID is enough for every compiled method (with inlining) ?

Reply:
These IDs are per compilation; ID=1 for the compilation of method `A.foo()` is not going to match ID=1 for the compilation of method `B.bar()`. Thus, @jdmpapin and I figure that 64k unique symbols should be sufficient for any compilation.

Reply:
64k should be enough (distinct symbols) for any one (AOT compilation) :sweat_smile:

Reply:
Okay the reason I asked was that it seemed like an ID gets assigned every time we compile time query something that is of relevance from a validation perspective. While the number of distinct symbols in a compilation may be well under 64k, it seemed to me that the number of IDs were more dependent on how many times we queried and I was wondering if in a large method with many passes of different optimizations we might get close to 64k queries. Is there some simple safety valve possible anyway to detect and fail the AOT compilation if the ID goes above 64k ?

Reply:
>While the number of distinct symbols in a compilation may be well under 64k, it seemed to me that the number of IDs were more dependent on how many times we queried 

That's not true, if we queried for class `B` from the constant pool of class `C` at cpIndex `5`, then regardless of how many times times we ran that query, once a class has been assigned an ID, it is unique to that ID; there is a 1-1 mapping between an ID and a `J9Class`. Additionally, there is only record per unique query made. So, as in the example above, even if the query was made 10 times, there would only be one validation record stored, since we only need to check once whether class `B` can be retrieved from the constant pool of class `C` at cpIndex `5`. However, if we retrieved class `B` from the constant pool of class `C` at cpIndices `3`, `4`, and `5`, then we'd need three validation records. but the validation records would look something like this (assuming class `B` <--> ID 1 and class `C` <--> ID 2):

ClassFromCP,ID=2,BeholderID=1,cpIndex=3
ClassFromCP,ID=2,BeholderID=1,cpIndex=4
ClassFromCP,ID=2,BeholderID=1,cpIndex=5

All that said, I do have a check to ensure that we don't overflow the IDs just in case: https://github.com/eclipse/openj9/pull/2919/commits/9a11e2fa6df3fafe57023d05e55df92042f2ba74#diff-3e7f7c3bf56a075860c3736280f74b05R108

Reply:
Okay thanks for the clarification and the check for overflow.